helm/hpcc/templates/esp.yaml

```diff
@@ -135,15 +141,23 @@ apiVersion: v1
 kind: Service
 metadata:
   name: {{ .name | quote }}
+{{ if hasKey .service "labels" -}}
+  labels:
+{{ toYaml .service.labels | indent 8 }}
+{{ end -}}
+{{- if hasKey .service "annotations" }}
+    annotations:
+{{ toYaml .service.annotations | indent 8 }}
+{{ end }}
 spec:
   ports:
-  - port: {{ required "servicePort must be specified" .servicePort }}
+  - port: {{ required "servicePort must be specified" .service.servicePort }}
     protocol: TCP
-    targetPort: {{ .port | default 8880 }}
+    targetPort: {{ .service.port | default 8880 }}
   selector:
     run: {{ .name | quote }}
-  type: {{ .public | ternary "LoadBalancer" "ClusterIP" }}
+  type: {{ .service.public | ternary "LoadBalancer" "ClusterIP" }}
```

Comment:
The whole of this Service generation before/now looks very generic, only thing that's specific to esp perhaps is the default ports.
It would be good to move it into _helpers.tpl, as a generateService call, and use in other places too.
(think it's same for example as on in Roxie apart from the default port aspect)

Reply:
Done